packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx

```diff
@@ -100,51 +101,86 @@ const FilterItem: FC<SchedulerFilterItemProps> = ({
 type SchedulerFiltersProps = {
     dashboard?: Dashboard;
     onChange: (schedulerFilters: SchedulerFilterRule[]) => void;
+    schedulerFilters: DashboardFilterRule[] | undefined;
 };
 
 const SchedulerFilters: FC<SchedulerFiltersProps> = ({
     dashboard,
-    // onChange,
+    schedulerFilters,
+    onChange,
 }) => {
     const { data: project, isLoading } = useProject(dashboard?.projectUuid);
-    // TODO: should read initial state from the BE
-    const [schedulerFilters, setSchedulerFilters] = useState<
-        DashboardFilterRule[]
-    >([]);
+    const isLoadingDashboardFilters = useDashboardContext(
+        (c) => c.isLoadingDashboardFilters,
+    );
+    const allFilters = useDashboardContext((c) => c.allFilters);
+    const fieldsWithSuggestions = useDashboardContext(
+        (c) => c.fieldsWithSuggestions,
+    );
+    const originalDashboardFilters = dashboard?.filters;
+
+    const [schedulerFiltersData, setSchedulerFiltersData] = useState<
+        DashboardFilterRule[] | undefined
+    >(schedulerFilters);
 
     const handleUpdateSchedulerFilter = useCallback(
         (schedulerFilter: SchedulerFilterRule) => {
-            // TODO: this should diff if the filter is actually
-            // different from the dashboard filter
-
-            const newState = produce(schedulerFilters, (draft) => {
-                const filterIndex = draft.findIndex(
-                    (f) =>
-                        f.target.fieldId === schedulerFilter.target.fieldId &&
-                        f.target.tableName === schedulerFilter.target.tableName,
-                );
+            if (!originalDashboardFilters) return;
 
-                if (draft[filterIndex]) {
-                    draft[filterIndex] = schedulerFilter;
-                } else {
-                    draft.push(schedulerFilter);
-                }
-            });
+            const originalFilter = originalDashboardFilters.dimensions.find(
+                (d) => d.id === schedulerFilter.id,
+            );
 
-            setSchedulerFilters(newState);
+            if (!originalFilter) return;
 
-            // TODO: sync with upper component
-            // call onChange - this can be debounced
-        },
-        [schedulerFilters],
-    );
+            let updatedFilters = schedulerFiltersData
+                ? [...schedulerFiltersData]
```

Comment:
Is there a place where we remove the invalid scheduler filters and then save? 
Looking at the code it looks like we don't show the invalid filters but they will keep poluting the database.

Reply:
great point, going to replicate that and find a solution. I think this is not handled yet!

Reply:
done: https://github.com/lightdash/lightdash/pull/7985/commits/7088c21f0c0364d08a963fdcadef9fd34b98b866

there's also the possibility that the user doesn't interact with the UI, but the scheduler keeps going. _old_/invalid filters should then be removed on the BE through some type of logic. But I think that can be handled on a different PR

Reply:
I've also reordered some conditional checks to make it more readable but still allow `schedulerFilters` to be `undefined` if no change etc etc [chore: reorder conditional checks on scheduler Filters](https://github.com/lightdash/lightdash/pull/7985/commits/58b4811894e910a1d5e194888185ecaae5524132)